misc/services/mobile-webui/mobile-webui-frontend/src/containers/activities/picking/PickStepScanScreen.jsx

```diff
@@ -7,109 +7,129 @@ import counterpart from 'counterpart';
-    const { workflowId: wfProcessId, activityId, lineId, stepId, appId, altStepId } = match.params;
+class PickStepScanScreen extends PureComponent {
+  constructor(props) {
+    super(props);
 
-    const wfProcess = selectWFProcessFromState(state, wfProcessId);
-    const stepProps = wfProcess.activities[activityId].dataStored.lines[lineId].steps[stepId];
-
-    return {
-      appId,
-      wfProcessId,
-      activityId,
-      lineId,
-      stepId,
-      altStepId,
-      stepProps,
-      qtyTarget: altStepId ? stepProps.pickFromAlternatives[altStepId].qtyToPick : stepProps.qtyToPick,
-      eligibleBarcode: altStepId
-        ? stepProps.pickFromAlternatives[altStepId].huBarcode
-        : stepProps.mainPickFrom.huBarcode,
+    this.state = {
+      scannedBarcode: null,
     };
-  };
+  }
 
-  class Wrapped extends PureComponent {
-    constructor(props) {
-      super(props);
+  onComponentDidMount() {
+    const { pushHeaderEntry } = this.props;
+    const location = pickingStepScanScreenLocation(this.props);
+    const headerHuCode = getPickFrom(this.props).huBarcode;
+    const headerQtyToPick = getQtyToPick(this.props).qtyPicked;
 
-      this.state = {
-        scannedBarcode: null,
-      };
-    }
+    pushHeaderEntry({
+      location,
+      values: [
+        {
+          caption: counterpart.translate('general.Barcode'),
+          value: headerHuCode,
+        },
+        {
+          caption: counterpart.translate('general.QtyToPick'),
+          value: headerQtyToPick,
+        },
+      ],
+    });
+  }
 
-    setScannedBarcode = (scannedBarcode) => {
-      this.setState({ scannedBarcode });
-    };
+  setScannedBarcode = (scannedBarcode) => {
+    this.setState({ scannedBarcode });
+  };
 
-    pushUpdatedQuantity = ({ qty = 0, reason = null }) => {
-      const { updatePickingStepQty, wfProcessId, activityId, lineId, stepId, go, altStepId, qtyTarget } = this.props;
-      const { scannedBarcode } = this.state;
-      const qtyRejected = qtyTarget - qty;
+  pushUpdatedQuantity = ({ qty = 0, reason = null }) => {
+    const { updatePickingStepQty, wfProcessId, activityId, lineId, stepId, go, altStepId, qtyTarget } = this.props;
+    const { scannedBarcode } = this.state;
+    const qtyRejected = qtyTarget - qty;
 
-      postStepPicked({
-        wfProcessId,
-        activityId,
-        stepId,
-        huBarcode: scannedBarcode,
-        qtyPicked: qty,
-        qtyRejectedReasonCode: reason,
-        qtyRejected,
+    postStepPicked({
+      wfProcessId,
+      activityId,
+      stepId,
+      huBarcode: scannedBarcode,
+      qtyPicked: qty,
+      qtyRejectedReasonCode: reason,
+      qtyRejected,
+    })
+      .then(() => {
+        updatePickingStepQty({
+          wfProcessId,
+          activityId,
+          lineId,
+          stepId,
+          altStepId,
+          qtyPicked: qty,
+          qtyRejected,
+          qtyRejectedReasonCode: reason,
+        });
+        go(-2);
```

Comment:
i think it would be helpful for future us to add here a comment about what's the expected screen to go for -2.

e..g. 
```javascript
go(-2); // go to picking line screen
```